### ISSUE TITLE
undefined variable not detected

### ISSUE THREAD

**Issue by doekia** [phpmd/phpmd#573]
Consider the following code:
```php
<?php

function not_detected($response){
        if (!empty($reponse->status)){
                print_r(array($reponse->status,$response->text));
        }
}

function not_detected_either($response){
        if ($reponse->status){
                print_r(array($reponse->status,$response->text));
        }
}

function detected($response){
        print_r(array($reponse->status,$response->text));
}
```

The $reponse (typo) is mainly undetected.
BTW, this tool rocks :1st_place_medal:

**Comment by kylekatarnls** [comment]
Hello, I just ran:
```
phpmd test.php text cleancode
```
Where **test.php** is the code you gave here.

And I get:

```
test.php:4     Avoid using undefined variables such as '$reponse' which will lead to PHP notices.
test.php:5     Avoid using undefined variables such as '$reponse' which will lead to PHP notices.
test.php:10    Avoid using undefined variables such as '$reponse' which will lead to PHP notices.
test.php:11    Avoid using undefined variables such as '$reponse' which will lead to PHP notices.
test.php:16    Avoid using undefined variables such as '$reponse' which will lead to PHP notices.
```

Which is the complete list of the typo occurrences.

Can you update and retry?

If you still have the issue. Could you also compare with the next version requiring `"phpmd/phpmd": "dev-master"`

Thank you.

**Comment by kylekatarnls** [comment]
As per my tests it should be solved since 2.8.0
